website/docs/cdktf/examples-and-guides/refactoring.mdx

```diff
@@ -0,0 +1,255 @@
+---
+page_title: Refactoring - CDK for Terraform
+description: >-
+  Refactor stacks as your CDKTF application grows in size and complexity.
+---
+
+# Refactoring
+
+You may need to move resources between stacks as your CDK for Terraform (CDKTF) application grows in size and complexity. This guide demonstrates how to refactor your stacks. In general you can change the names of stacks freely without an effect on the synthesized code. If you change the name of a resource this might lead to a re-creation of the resource, especially if you move the resource from one stack to another.
+
+## Moving resources inside a stack
+
+Let's say you have a stack deployed with an aws `S3Bucket` resource and you want to give it another name:
+
+```typescript
+import { Construct } from "constructs";
+import { App, TerraformStack } from "cdktf";
+import { AwsProvider } from "@cdktf/provider-aws/lib/provider";
+import { S3Bucket } from "@cdktf/provider-aws/lib/s3-bucket";
+
+class MyStack extends TerraformStack {
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    new AwsProvider(this, "aws", {
+      region: "us-east-1",
+    });
+
+    // Old S3Bucket definition
+    // new S3Bucket(this, "my-bucket", {
+    //   bucketPrefix: "my-bucket",
+    // });
+
+    // New S3Bucket definition
+    new S3Bucket(this, "new-bucket", {
+      bucketPrefix: "my-bucket",
+    });
+  }
+}
+
+const app = new App();
+new MyStack(app, "refactoring-example");
+app.synth();
+```
+
+When you run a `cdktf plan` you might see the following output:
```

Comment:
We want to stay away from "see" to promote inclusion (per our style guide). I'm going to go through and suggest some alternatives, but just a note for future!

```suggestion
When you run a `cdktf plan`, CDKTF displays the following output:
```